### DISCUSSION TITLE
AIMP from official site

### DISCUSSION THREAD

**Discussion by eXense-auth** [ImranR98/Obtainium#1675] [Q&A]
please help me to add AIMP player from official site 
https://www.aimp.ru/?do=download&os=android

I used to have everything working with version extraction, but then they stopped and I don't know the reason why 

my old config for stable release:
https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22com.aimp.player%22%2C%22url%22%3A%22https%3A%2F%2Fwww.aimp.ru%2F%3Fdo%3Ddownload%26os%3Dandroid%22%2C%22author%22%3A%22www.aimp.ru%22%2C%22name%22%3A%22AIMP%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22do%3Ddownload.file%26id%3D13%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Atrue%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22AIMP%20(v%5C%5C%5C%5Cd%2B%5C%5C%5C%5C.%5C%5C%5C%5Cd%2B%5C%5C%5C%5C.%5C%5C%5C%5Cd%2B)%3C%2Fh1%3E%3Ch2%3E%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%241%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%26id%3D22%20-%20beta%2C%20RC%2C%2013%20-%20release%2C%2020%20-%20NoSAF%5C%22%7D%22%7D

**Comment by ImranR98** [comment]
Couldn't get it to work with the HTML source, but since the actual APK link is fixed, you could just add it with the "direct APK link" source. Tried that and it seems to work.

**Reply by eXense-auth** [reply]
yes, its worked, but without version number

**Reply by ImranR98** [reply]
Yes the version is in the actual filename which the app doesn't support extracting from right now. But that goes for the HTML source too.

**Reply by DarkCrypt** [reply]
The above config imported app as current version for me.

**Edit:**
I can also confirm the config is able to retrieve updates.

**Reply by eXense-auth** [reply]
> The above config imported app as current version for me.
> 
> **Edit:** I can also confirm the config is able to retrieve updates.

yes, but next release may not be updated

please check config for beta release 

https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22com.aimp.player%22%2C%22url%22%3A%22https%3A%2F%2Fwww.aimp.ru%2F%3Fdo%3Ddownload%26os%3Dandroid%22%2C%22author%22%3A%22www.aimp.ru%22%2C%22name%22%3A%22AIMP%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22intermediateLink%5C%22%3A%5B%5D%2C%5C%22customLinkFilterRegex%5C%22%3A%5C%22do%3Ddownload.file%26id%3D22%5C%22%2C%5C%22filterByLinkText%5C%22%3Afalse%2C%5C%22skipSort%5C%22%3Afalse%2C%5C%22reverseSort%5C%22%3Afalse%2C%5C%22sortByLastLinkSegment%5C%22%3Afalse%2C%5C%22versionExtractWholePage%5C%22%3Atrue%2C%5C%22requestHeader%5C%22%3A%5B%7B%5C%22requestHeader%5C%22%3A%5C%22User-Agent%3A%20Mozilla%2F5.0%20(Linux%3B%20Android%2010%3B%20K)%20AppleWebKit%2F537.36%20(KHTML%2C%20like%20Gecko)%20Chrome%2F114.0.0.0%20Mobile%20Safari%2F537.36%5C%22%7D%5D%2C%5C%22defaultPseudoVersioningMethod%5C%22%3A%5C%22APKLinkHash%5C%22%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22AIMP%20(v%5C%5C%5C%5Cd%2B%5C%5C%5C%5C.%5C%5C%5C%5Cd%2B%5C%5C%5C%5C.%5C%5C%5C%5Cd%2B%20%5C%5C%5C%5Cw%2B)%3C%2Fh1%3E%3Ch2%3E%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%241%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Afalse%2C%5C%22appName%5C%22%3A%5C%22%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22%26id%3D22%20-%20beta%2C%20RC%2C%2013%20-%20release%2C%2020%20-%20NoSAF%5C%22%7D%22%2C%22overrideSource%22%3Anull%7D

**Reply by DarkCrypt** [reply]
> yes, but next release may not be updated

Confirming the stable config retrieved a 2nd update. (v4.11.1475 -> v4.11.1476)

I'll test your Beta config.

**Edit:**
Your Beta config imports app but error "Could not get ID from APK" occurs when attempting to download.

**Reply by eXense-auth** [reply]
> > yes, but next release may not be updated
> 
> Confirming the stable config retrieved a 2nd update. (v4.11.1475 -> v4.11.1476)
> 
> I'll test your Beta config.
> 
> **Edit:** Your Beta config imports app but error "Could not get ID from APK" occurs when attempting to download.

yes, the same thing can happen with the stable version of aimp, I don't know what it depends on, for example, an earlier version did not download, giving the same error on the same configuration
